app/controllers/declaration/DeclarationHolderController.scala

```diff
@@ -66,16 +64,19 @@ class DeclarationHolderController @Inject()(
 
   private def addAnotherYesNoForm: Form[YesNoAnswer] = YesNoAnswer.form(errorKey = "declaration.declarationHolders.add.another.empty")
 
-}
-
-object DeclarationHolderController {
+  private def nextPageWhenNoHolders(implicit request: JourneyRequest[_]): Mode => Call =
+    if (request.declarationType == SIMPLIFIED) routes.DeclarationHolderAddController.displayPage
+    else routes.DeclarationHolderRequiredController.displayPage
 
-  def nextPage(implicit request: JourneyRequest[_]): Mode => Call =
+  private def nextPage(implicit request: JourneyRequest[_]): Mode => Call =
     request.declarationType match {
-      case SUPPLEMENTARY | STANDARD =>
-        controllers.declaration.routes.OriginationCountryController.displayPage
-      case SIMPLIFIED | OCCASIONAL | CLEARANCE =>
-        controllers.declaration.routes.DestinationCountryController.displayPage
+      case SUPPLEMENTARY | STANDARD => controllers.declaration.routes.OriginationCountryController.displayPage
+      case _                        => controllers.declaration.routes.DestinationCountryController.displayPage
     }
+}
+
+object DeclarationHolderController {
 
+  def cachedHolders(implicit request: JourneyRequest[_]): Seq[DeclarationHolder] =
+    request.cacheModel.parties.declarationHoldersData.getOrElse(DeclarationHoldersData.empty).holders
```

Comment:
The idea behind extracting this method and sharing it between different controllers is great. But I have 2 comments regarding it:
1. Could this method be more like `request.cacheModel.parties.declarationHoldersData.map(_.holders).getOrElse(Seq.empty)` ? I am concerned about using `DeclarationHoldersData.empty` here. It creates another place to look at when reading the code. 
2. I understand that putting this method inside companion object is the quickest way, but maybe you could create a separate `object` for that? This way it would be more clear that this method can be used by all related classes.

Disclaimer: It is probably nit-picking as I usually do, but I must admit that I don't feel comfortable with the current shape of that method. Happy to discuss if anyone has different perspective.